### DISCUSSION TITLE
What are the recommended distros to use when installing on vcluster?

### DISCUSSION THREAD

**Discussion by aqeelat** [deployKF/deployKF#107] [Questions & Answers] [answered]
The docs suggest using vcluster, but not what distro to use with it. vCluster supports k8s, k3s, and k0, with the default version being k3s.

**Comment by thesuperzapper** [comment, marked as answer]
Personally, I always use K3S, and that seems to be the [recommended distribution from vCluster themselves](https://www.vcluster.com/docs/deploying-vclusters/supported-distros). But I'm sure the others will work correctly too, so I guess it's a matter of personal preference or business requirements.

---

As a side note, if you plan to use vCluster, you will probably want to know about the following values:

- [`argocd.appNamePrefix`](https://github.com/deployKF/deployKF/blob/v0.1.4/generator/default_values.yaml#L8-L13): which will let you have a single ArgoCD manage multiple deployKF instances
- [`argocd.destination`](https://github.com/deployKF/deployKF/blob/v0.1.4/generator/default_values.yaml#L56-L61): which will let you install deployKF on a separate cluster to the one running ArgoCD itself

These values are important depending on where you decide to run your ArgoCD cluster, either a single one in the host cluster, or inside each vCluster.

___EDIT:__ I have just [updated the docs](https://www.deploykf.org/guides/dependencies/argocd/#can-a-single-argo-cd-deploy-multiple-clusters) to explain those values, and [updated](https://github.com/deployKF/deployKF/pull/108) the [`sync_argocd_apps.sh`](https://github.com/deployKF/deployKF/blob/main/scripts/sync_argocd_apps.sh) script to support syncing only the apps which have a specific `argocd.appNamePrefix`._
